litellm/integrations/opik/opik.py

```diff
@@ -0,0 +1,273 @@
+
+            if len(self.log_queue) >= self.batch_size:
+                verbose_logger.debug("OpikLogger - Flushing batch")
+                await self.flush_queue()
+        except Exception as e:
+            verbose_logger.exception(
+                f"OpikLogger failed to log success event - {str(e)}\n{traceback.format_exc()}"
+            )
+    
+    def log_success_event(self, kwargs, response_obj, start_time, end_time):
+        try:
+            opik_payload = self._create_opik_payload(
+                kwargs=kwargs,
+                response_obj=response_obj,
+                start_time=start_time,
+                end_time=end_time
+            )
+
+            self.log_queue.extend(opik_payload)
+            verbose_logger.debug(f"OpikLogger added event to log_queue - Will flush in {self.flush_interval} seconds...")
+
+            if len(self.log_queue) >= self.batch_size:
+                verbose_logger.debug("OpikLogger - Flushing batch")
+                self.flush_queue()
+        except Exception as e:
+            verbose_logger.exception(
+                f"OpikLogger failed to log success event - {str(e)}\n{traceback.format_exc()}"
+            )
+
+    async def _submit_batch(self, url: str, headers: Dict[str, str], batch: List[Dict]):
+        try:
+            response = await self.async_httpx_client.post(
+                url=url,
+                headers=headers,
+                json=batch
+            )
+            response.raise_for_status()
+
+            if response.status_code >= 300:
+                verbose_logger.error(
+                    f"OpikLogger - Error: {response.status_code} - {response.text}"
+                )
+            else:
+                verbose_logger.debug(
+                    f"OpikLogger - {len(self.log_queue)} Opik events submitted"
+                )
+        except Exception as e:
+            verbose_logger.exception(
+                f"OpikLogger failed to send trace batch - {str(e)}\n{traceback.format_exc()}"
+            )
+
+    async def async_send_batch(self):
+        verbose_logger.exception("Calling async_send_batch")
+        if not self.log_queue:
+            return
+        
+        trace_url = f"{self.opik_base_url}/v1/private/traces/batch"
+        span_url = f"{self.opik_base_url}/v1/private/spans/batch"
+
+        # Create headers
+        headers = {}
+        if self.opik_workspace:
+            headers["Comet-Workspace"] = self.opik_workspace
+        
+        if self.opik_api_key:
+            headers["authorization"] = self.opik_api_key
+
+        # Split the log_queue into traces and spans
+        def remove_nulls(x):
+            x_ = {k:v for k,v in x.items() if v is not None}
+            return x_
+
+        traces = [remove_nulls(x) for x in self.log_queue if "type" not in x]
+        spans = [remove_nulls(x) for x in self.log_queue if "type" in x]
```

Comment:
Can you use better variable names than x, x_, k and v? Also it's not clear when reading just this piece of code how actually you distinguish between traces and spans? Does a span is a dict with the key "type" and traces are everything else?